src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -569,5 +569,12 @@ static void AppendHexDumpRowPrefix(StringBuilder dump, int row, int rowStartInde
                 dump.Append('|');
             }
         }
+
+        public static string DecodeString(IByteBuffer src, int readerIndex, int len, Encoding encoding)
+        {
+            if (len == 0) return string.Empty;
+            
+            return encoding.GetString(src.Array.Slice(len));
```

Comment:
@nayato I think that this case is not necessary to allocate new ByteBuffer. If ```buf.HasArray==false```, just return string.Empty.
So i understand like following code.
How do you think about this code??
```C#
if (src.HasArray == false || len == 0)
{
	return string.Empty;
}
return encoding.GetString(src.Array, readerIndex, len);
```